src/Abp.ZeroCore.EntityFrameworkCore/EntityHistory/EntitySnapshotManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Abp.Dependency;
+using Abp.Domain.Entities;
+using Abp.Domain.Repositories;
+using Abp.EntityFramework;
+using Abp.EntityFrameworkCore.Repositories;
+using Abp.Events.Bus.Entities;
+using Abp.Json;
+using Microsoft.EntityFrameworkCore;
+
+namespace Abp.EntityHistory
+{
+    public class EntitySnapshotManager : IEntitySnapshotManager, ITransientDependency
+    {
+        private readonly IRepository<EntityChange, long> _entityChangeRepository;
+
+        public EntitySnapshotManager(IRepository<EntityChange, long> entityChangeRepository)
+        {
+            _entityChangeRepository = entityChangeRepository;
+        }
+        public async Task<EntityHistorySnapshot> GetSnapshotAsync<TEntity>(int id, DateTime snapshotTime) where TEntity : class, IEntity<int>
+        {
+            return await GetSnapshotAsync<TEntity, int>(id, snapshotTime);
+        }
+        public async Task<EntityHistorySnapshot> GetSnapshotAsync<TEntity, TPrimaryKey>(TPrimaryKey id, DateTime snapshotTime) where TEntity : class, IEntity<TPrimaryKey>
+        {
+            var entity = _entityChangeRepository.GetDbContext()
+                .Set<TEntity>().AsQueryable().FirstOrDefault(x => x.Id.Equals(id));
+
+            var snapshotPropertiesDictionary = new Dictionary<string, string>();
+            var propertyChangesStackTreeDictionary = new Dictionary<string, string>();
+
+            if (entity != null)
+            {
+                string fullName = typeof(TEntity).FullName;
+                var idJson = id.ToJsonString();
+
+                var allChanges = _entityChangeRepository.GetAll().Include(x => x.PropertyChanges).ToList();
```

Comment:
seems like we don't need `allChanges`?

Reply:
a forgotten control code, i will delete 👍